public/js/Filter.js

```diff
@@ -120,21 +139,14 @@ function SampleFilter(uniqueRow, uniqueCol, uniquePlate, uniqueGene) {
     $('#filter-button').removeClass('hidden');
 }
```

Comment:
This function is enormous! Can you manage to break it down a bit, farm out a few of those loops to some helper functions? (and I think this closing curly brace is misaligned LOL)